keps/sig-architecture/5241-beta-featuregate-promotion-requirements/README.md

```diff
@@ -0,0 +1,246 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Features gates must include all functional, security, and testing requirements along with
+resolving all issues and gaps identified prior to being enabled by default.
+The only valid GA criteria are “all issues and gaps identified as feedback during beta are resolved”.
+
+## Motivation
+
+Features gates that are enabled by default are enabled in every production Kubernetes cluster in the world.
+We must avoid making every production cluster into unstable or incomplete feature testing clusters.
+Even feature gates that make flags accessible, but require a secondary configuration to use must be
+stable, because it is unrealistic to expect everyone to understand the graduation stages of various flags
+for each release: the only stages that really matter are "takes enabling an explicit alpha feature gate"
+and "my production cluster accepts this as valid by default".
+
+### Goals
+
+* Features gates must include all functional, security, and testing requirements along with
+  resolving all issues and gaps identified prior to being enabled by default.
+* The only valid GA criteria are “all issues and gaps identified as feedback during beta are resolved”.
+
+### Non-Goals
+
+* Changing beta APIs off by default rules.
+* Change the imperfect mechanisms we have for API evolution.
+
+## Proposal
+
+Kubernetes feature gates have three levels: GA, Beta, and Alpha.
+1. GA means that a feature gate is unconditionally enabled in all production kubernetes clusters and
+   that feature cannot be disabled.
+2. Beta means that a feature gate is usually enabled in all production Kubernetes clusters by default
+   and that feature can be disabled.
+   Exceptions exist for entirely new APIs and some node features, but this broadly the case.
```

Comment:
this tends to confuse readers and new developers not familiar with the the duality of API version v1alpha1, v1beta1, v1 / feature gate alpha,beta, GA and the coupling of both lifecycles. Since the restriction about beta apis disabled by default, that makes feature gates that are beta and depende on api beta versions disabled by default.
`Beta features non dependent on beta APIs are usually enabled by default` ??? I do not know well how to express this conundrum

Reply:
Adding the detail about how we have the GA and disable-able state helped I think.